src/toxcore/dht/codec.rs

```diff
@@ -33,22 +90,95 @@ pub enum DecodeError {
     },
     /// Error indicates that received packet can't be parsed
     #[fail(display = "Deserialize Packet error: {:?}, packet: {:?}", error, packet)]
-    DeserializeError {
+    Deserialize {
         /// Parsing error
         error: ErrorKind,
         /// Received packet
         packet: Vec<u8>,
+    },
+    /// For From<DecodeError> trait.
+    #[fail(display = "Error: {}", msg)]
+    String {
```

Comment:
String -> Custom ?

Reply:
Also why don't you keep IoError inside the enum?

Reply:
It's for error from tokio, maybe just call it `IoError`? `Custom` is not very explanatory.